pkg/oci/load_balancer_security_lists.go

```diff
@@ -390,16 +435,38 @@ func portInUse(serviceLister listersv1.ServiceLister, port int32) (bool, error)
 	return false, nil
 }
 
+func healthCheckPortInUse(serviceLister listersv1.ServiceLister, port int32) (bool, error) {
```

Comment:
Why do you need this? Can health checks share ports when set on the spec? I thought they were randomized just like node ports and unique.

Reply:
If 2 services were using the default health check port (10256 on kube-proxy), then wouldn't they be sharing the same rule in both the load balancer egress and node ingress?

Reply:
That's never going to be set on the load balancer spec afaik. The health check ports have to be in the node port range, which 10256 does not fall into. If you look through the code of the various providers, including ours, you'll notice it's set specifically.

Reply:
Yep right. I guess the implementation of healthCheckPortInUse is functionally correct though, i.e. in the case where the default port is used we must still search over the other services to see if any of them use the default port too. However, it can definitely be optimised in that there's no need to look at the other services if a custom health check port is passed (because as you said, we know they will be unique). Therefore it could look something like the following:
```
func healthCheckPortInUse(serviceLister listersv1.ServiceLister, port int32) (bool, error) {
	if port != lbNodesHealthCheckPort {
		// This service is using a custom healthcheck port (enabled through setting
		// extenalTrafficPolicy=Local on the service). As this port is unique
		// per service, we know no other service will be using this port too.
		return false, nil
	}

	// This service is using the default healthcheck port, so we must check if
	// any other service is also using this default healthcheck port.
	serviceList, err := serviceLister.List(labels.Everything())
	if err != nil {
		return false, err
	}
	for _, service := range serviceList {
		if service.Spec.Type == api.ServiceTypeLoadBalancer {
			healthCheckPath, _ := apiservice.GetServiceHealthCheckPathPort(service)
			if healthCheckPath == "" {
				// We have found another service using the default port.
				return true, nil
			}
		}
	}
	return false, nil
}
```
How does this look to you? Is this closer to what you had in mind, or am I totally missing the point here?